src/Discovery/DiscoveryDetails.tsx

```diff
@@ -127,17 +166,30 @@ const accessDescriptor = (resource: DiscoveryResource) => {
 type TabFieldConfig = TabFieldGroup['fields'][0]
 type TabFieldGroup = DiscoveryConfig['detailView']['tabs'][0]['groups'][0];
 
-const formatResourceValuesWhenNestedArray = (resourceFieldValue: string[]) => {
+const formatResourceValuesWhenNestedArray = (resourceFieldValue) => {
   if (
     Array.isArray(resourceFieldValue)
-    && Array.isArray(resourceFieldValue[0])
   ) {
-    return resourceFieldValue[0].join(', ');
+    if (Array.isArray(resourceFieldValue[0]) && resourceFieldValue[0].every((val) => typeof val === 'string')) {
+      return resourceFieldValue[0].join(', ');
+    }
+    return resourceFieldValue[0];
   }
   return resourceFieldValue;
 };
 
 const tabField = (fieldConfig: TabFieldConfig, discoveryConfig: DiscoveryConfig, resource: DiscoveryResource): JSX.Element => {
+  // Setup special fields first
+  if (fieldConfig.type === 'accessDescriptor') {
+    return accessDescriptor(resource);
+  }
+  if (fieldConfig.type === 'tags') {
+    const tags = fieldConfig.categories ? (resource.tags || []).filter(
```

Comment:
Consider breaking this apart into a separate function for clarity.


 For example: 
```
const filterTagsByCategories = (fieldConfig: TabFieldConfig, resource: DiscoveryResource) => {
  const { categories } = fieldConfig;
  const { tags } = resource;  
  if (categories && Array.isArray(tags)) {
    const filteredTags = tags.filter(tag => categories.includes(tag.category));
    return filteredTags;
  }
  return tags;
}

const tags = filterTagsByCategories(fieldConfig, resource);
```

Note: code is not tested.

Reply:
I think this is because the intention of the developer at the time of implementing this originally was to re-use the `renderFieldContent()` function from the main discovery page. With that being said, I'm not objecting to move this logic out to make it a component on its own, but I think that should be the part of the work of the new "modularizing disco details page components" ticket